server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithoutValueFilterDataSet.java

```diff
@@ -70,9 +71,10 @@ public GroupByWithoutValueFilterDataSet(long queryId, List<Path> paths, long uni
   /**
    * init reader and aggregate function.
    */
-  public void initGroupBy(QueryContext context, List<String> aggres, IExpression expression)
+  public void initGroupBy(QueryContext context, List<String> aggres, List<TSDataType> dataTypes,
```

Comment:
Change this to a private method and called in constructor. More, the constructor could receive a GroupByPlan as a parameter, which is clearer